### PR TITLE
Fixed minor bugs with Steam Washer

### PR DESCRIPTION
**Pull request by evgengoldwar**
Added mode change to the controller's gui

Added the ability for Steam Washer to fill water. Like Washer

### CODE REVIEW

File: src/main/java/gtPlusPlus/xmod/gregtech/common/tileentities/machines/multi/processing/steam/GregtechMetaTileEntity_SteamWasher.java [hunk 1]
```diff
@@ -432,4 +456,99 @@ public void setMachineModeIcons() {
...
+                xOffset += 4;
+            } else xOffset -= 6;
+        }
+
+        if (frontFacing == ForgeDirection.SOUTH) {
+            zOffset -= 1;
+            if (curFlip.isHorizontallyFlipped()) {
+                xOffset -= 6;
+            } else xOffset += 4;
+        }
```
Comment (Nxer) [inline src/main/java/gtPlusPlus/xmod/gregtech/common/tileentities/machines/multi/processing/steam/GregtechMetaTileEntity_SteamWasher.java:508]:
```
        if (frontFacing == ForgeDirection.WEST) {
            xOffset -= 1;
            if (curFlip.isHorizontallyFlipped()) {
                zOffset -= 6;
            } else zOffset += 4;
        } else if (frontFacing == ForgeDirection.EAST) {
            xOffset -= 1;
            if (curFlip.isHorizontallyFlipped()) {
                zOffset += 4;
            } else zOffset -= 6;
        } else if (frontFacing == ForgeDirection.NORTH) {
            zOffset -= 1;
            if (curFlip.isHorizontallyFlipped()) {
                xOffset += 4;
            } else xOffset -= 6;
        } else if (frontFacing == ForgeDirection.SOUTH) {
            zOffset -= 1;
            if (curFlip.isHorizontallyFlipped()) {
                xOffset -= 6;
            } else xOffset += 4;
        }
```
Use `else if` to separate `frontFacing` like using `switch`

**Review by Nxer — CHANGES_REQUESTED** [review]
Better to choose a timing to confirm the water block has been set, instead check water block in every recipe checking.

File: src/main/java/gtPlusPlus/xmod/gregtech/common/tileentities/machines/multi/processing/steam/GregtechMetaTileEntity_SteamWasher.java (hunk 1 above)
Comment (evgengoldwar) [reply src/main/java/gtPlusPlus/xmod/gregtech/common/tileentities/machines/multi/processing/steam/GregtechMetaTileEntity_SteamWasher.java]: Fixed

**Comment by Nxer** [comment]
image
Something wrong in structure definition.

**Review by Nxer — APPROVED** [review]
No problem overall.

**Pull request merged**